ivy/functional/backends/paddle/elementwise.py

```diff
@@ -578,75 +430,64 @@ def acos(x: paddle.Tensor, /, *, out: Optional[paddle.Tensor] = None) -> paddle.
     return paddle.acos(x)
 
 
-@with_unsupported_device_and_dtypes(
-    {"2.5.1 and below": {"cpu": ("complex64", "complex128")}},
+@with_supported_dtypes(
+    {
+        "2.5.1 and below": (
+            "bool",
+            "int8",
+            "int16",
+            "in32",
+            "in64",
+            "float",
```

Comment:
is `float16` supported?
also, can we get `complex64` and `complex128` working for this function?

Reply:
no `float16` does not work

Reply:
yeah I can add the complex support.